iroha/src/isi.rs

```diff
@@ -189,12 +194,30 @@ impl Execute for MintBox {
                 Value::U32(quantity) => {
                     Mint::<Asset, u32>::new(quantity, asset_id).execute(authority, world_state_view)
                 }
-                _ => Err(error!("Unsupported instruction.").into()),
+                Value::Bool(..)
+                | Value::String(..)
+                | Value::Vec(..)
+                | Value::Id(..)
+                | Value::Identifiable(..)
+                | Value::PublicKey(..)
+                | Value::Parameter(..)
+                | Value::SignatureCheckCondition(..)
+                | Value::TransactionValue(..)
+                | Value::PermissionToken(..) => Err(error!("Unsupported instruction.").into()),
```

Comment:
This becomes unreadable... I don't think we need this restriction, Generally `_` match is a good pattern to my mind.

Reply:
Motivation for this pattern is not to miss updates for enums. If there will be new variants we won't be able to compile it and need to manually recheck this code.

Reply:
Yes, I understand, but in our codebase we usually match as you can see here only against a couple of values, and now we have to specify the whole enum.

Reply:
I do believe that it increases the perceived complexity of the code and decreases readability.